bigtable/client/row_set.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_ROW_SET_H_
+#define GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_ROW_SET_H_
+
+#include "bigtable/client/version.h"
+
+#include <google/bigtable/v2/data.pb.h>
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+/**
+ * Temporary definition of the RowSet class.
+ *
+ * TODO(#31): Replace all this with a full implementation.
+ */
+class RowSet {
+ public:
+  /// All rows.
+  RowSet() : row_set_() {}
+
+  /// Clip by discarding all row keys up to and including this one.
+  void ClipUpTo(absl::string_view row_key) {
```

Comment:
Would you be Okay with a function that returns the new `RowSet`?  And what about `RemoveLessOrEqual`?  Or `KeepOnlyGreater`?  I am going for a name that is not specific to scans.

Reply:
Sure, I only put this as a placeholder.

To me, the nicest would be an `Intersect(RowRange r)`, but I'm fine with anything that works.

Reply:
For what is worth, I have a branch where I am working on `Intersect()`:

https://github.com/coryan/google-cloud-cpp/tree/implement-row-set-intersect

Let me know if that looks like what you need.  That branch will be the basis of a PR I will send your way, I need to write more tests though.  `RowSet::Intersect()` could be optimized too, specially in the case where there is just one range, or the range does not change (which I think will be common).

Reply:
Thanks for doing this (and for letting me know). I think it's looking great.